src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -164,6 +166,14 @@
         <UseAbsolutePath Condition="'$(TestWithLocalLibraries)'=='true'">$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').StartsWith('$(BinDir)'))</UseAbsolutePath>
       </IncludedFileForRunnerScript>
     </ItemGroup>
+    
+    <MakeDir Directories="$(OutputFolderForTestDependencies)" />
+    <Message Text="Generating $(OutputFolderForTestDependencies)/$(AssemblyName).dependencylist.txt" />
+    <WriteLinesToFile
+      File="$(OutputFolderForTestDependencies)/$(AssemblyName).dependencylist.txt"
```

Comment:
AssemblyName may actually be too general here, especially in cases were we build the same test project multiple times in the same build. You likely need to make it take other things into account, similar to what we do here https://github.com/dotnet/corefx/blob/master/dir.props#L497 https://github.com/dotnet/corefx/blob/master/dir.props#L498